_episodes/05-function-style.md

```diff
@@ -48,17 +51,90 @@ array(['O', 'H', 'H'], dtype='<U1')
 You can also see the atomic coordinates by executing:
 ~~~
 >>> coords
 ~~~
 {: .language-python}
 
-Hooray! It seems like this function works! However, our function might be hard to read and understand for others so we might want to consider styling it properly.
+Hooray! It seems like this function works! This should come as no surprise since we are the authors of the function and we know its internal structure. This is not necessarily true for someone editing our code and specially not true for someone just using our code. There are instances where even though the code is executed correctly, i.e., there where no syntax errors, an unwanted expected behavior occurs. In these cases, our code should be able to stop itself to prevent a malfunction. 
+
+## Raising Errors 
+
+Take for example the division by zero. If we try to calculate 
+~~~
+>>> 1/0
+~~~
+{: .language-python}
+
+We would get
+
+~~~
+ZeroDivisionError: division by zero
+~~~
+{: .output}
+
+In this example, the code was smart enough to identify the division by zero and halted. This type of feedback is much more helpful than just throwing an ugly `NaN`. This is called an exception error. There are several built-in exception such as the "ZeroDivisionError", but in general one can simply use a `raise Exception`. Let us see how we could accomplish this. 
+
+Consider our function `calculate distance.py`
+
+~~~
+def calculate_distance(rA, rB):
+    dist_vec = (rA - rB)
+    distance = np.linalg.norm(dist_vec)
+    return distance
+~~~
+{: .language-python}
+
+The function returns a distance no mater what the two points are. Now, imagine that an xyz file contained a repeated line, our function would do its job and print a distance equal to `0.0`. This is clearly unphysical and could lead into plotting two atoms in the same point. This is a perfect place to raise an exception. The syntax is as follows
+
+~~~
+def calculate_distance(rA, rB):
+    dist_vec = (rA - rB)
+    distance = np.linalg.norm(dist_vec)
+    if distance == 0.0:
+        raise Exception("Two atoms are located in the same point in space")
+    return distance
+~~~
+{: .language-python}
+
+In this way, trying to calculate the distance between the same two points would now show
+
+~~~
+Exception: Two atoms are located in the same point in space
+~~~
+{: .output}
+
+This is a general exception that requires a very explicit line to describe what the problem is. The already built-in exceptions include errors that are common while programming. For example, our function requires explicit use of numpy arrays. Nevertheless, a user may be tempted to use a list of length 3 to describe the position of two atoms. We know that it is not possible to perform arithmetic between full lists. In this case we might use the exception type `TypeError`
+
+~~~
+def calculate_distance(rA, rB):
+    if isinstance(rA,np.ndarray) or isinstance(rB,np.ndarray) is False:
```

Comment:
For this `if` statement, I believe you will want `(isinstance(rA,np.ndarray) is False) or (isinstance(rB,np.ndarray) is False). What is currently written will do the opposite of what you intend for the first statement